domains/oidc-mock/API/Mock/Controllers/AuthController.cs

```diff
@@ -49,23 +49,27 @@ public IActionResult Authorize(string client_id, string redirect_uri)
 
     [HttpPost]
     [Route("Connect/Token")]
-    public IActionResult Token(string grant_type, string code, string redirect_uri)
+    public IActionResult Token(string grant_type, string code, string? client_id, string? client_secret, string redirect_uri)
     {
-        var authorizationHeader = AuthenticationHeaderValue.Parse(Request.Headers["Authorization"]!);
-        logger.LogDebug("connect/token: authorization header: {AuthorizationHeader}", $"{authorizationHeader.Scheme} {authorizationHeader.Parameter}");
+        if (client_id is null || client_secret is null)
+        {
+            var authorizationHeader = AuthenticationHeaderValue.Parse(Request.Headers["Authorization"]!);
+            logger.LogDebug("connect/token: authorization header: {AuthorizationHeader}", $"{authorizationHeader.Scheme} {authorizationHeader.Parameter}");
+            var auth = (authorizationHeader.Parameter ?? ":").DecodeBase64();
+            var split = auth.Split(":");
```

Comment:
Seems a bit confusing :) Just put a few words in comment?

Reply:
Sure, since i just copied this i'll let the original auther add a comment :-D